src/Models/Nonhydrostatic2DModels/equations_pressure.jl

```diff
@@ -21,3 +23,29 @@ end
 
     return p
 end
+
+@inline function calculate_pressure(
+    Y,
+    Ya,
+    ::TotalEnergy,
+    ::EquilibriumMoisture,
+    params,
+    Φ,
+    K,
+    FT,
+)
+    ρ = Y.base.ρ
+    ρe_tot = Y.thermodynamics.ρe_tot
+    ρq_tot = Y.moisture.ρq_tot
+    e_int = @. ρe_tot / ρ - K - Φ
```

Comment:
Can we use the caching system here?

Reply:
Right. Thats my next todo item. But there is no cache set up there yet (despite Ya appearing here and there). So I want to add it in the next PR